crud.py

```diff
@@ -47,15 +48,19 @@ def create_exercise(db: Session, exercise: schemas.ExerciseCreate):
         text=exercise.text,
         complexity=exercise.complexity,
     )
-    db.add(db_exercise)
-    if exercise.category:
-        for category in exercise.category:
-            db_category = get_category(db, category)
-            if db_category is None:
-                db_category = create_category(db, category)
-            db_exercise.category.append(db_category)
-    db.commit()
-    db.refresh(db_exercise)
+    try:
+        db.add(db_exercise)
+        if exercise.category:
+            for category in exercise.category:
+                db_category = get_category(db, category)
+                if db_category is None:
+                    db_category = create_category(db, category)
+                db_exercise.category.append(db_category)
+        db.commit()
+        db.refresh(db_exercise)
+    except exc.SQLAlchemyError as e:
+        db.rollback()
+        raise
```

Comment:
What error comes from here? If we are catching it and doing a rollback, we should probably return it to `main.py` and let the user know that an error occurs, if we just `raise` the app will crash and the user will get `500: Internal Server Error`

Reply:
I add it when I had an error with the tables SQL (after I changed a model)